test/e2e/validation_test.go

```diff
@@ -178,19 +146,13 @@ func (tc *testContext) getCredsFromSecret(instanceID string) (wmc.Credentials, e
 
 // validateInstanceSecret validates the instance secret.
 func (tc *testContext) validateInstanceSecret(instanceID string) error {
-	ipAddress, err := tc.getNodeIP(instanceID)
-	if err != nil {
-		return err
-	}
-	windowsVM, err := tc.getWindowsVM(ipAddress, instanceID)
+	creds, err := tc.getCredsFromSecret(instanceID)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "error while getting creds from secret")
 	}
-	err = tc.validateConnectivity(windowsVM)
-	if err != nil {
-		return err
+	if creds == (wmc.Credentials{}) {
```

Comment:
is there a reason why we check it against wmc.Credentials{} but not nil?

Reply:
Creds isnt a pointer so it cannot be nil, the empty wmc.Credentials{} struct is its default value.